netmaster/master/netmaster.go

```diff
@@ -283,15 +305,17 @@ func stopAndRemoveServiceContainer(tenantName string) error {
 
 // DeleteTenantID deletes a tenant from the state store, by ID.
 func DeleteTenantID(stateDriver core.StateDriver, tenantID string) error {
-	if GetClusterMode() == "docker" {
-		err := stopAndRemoveServiceContainer(tenantID)
+	var err error
+
+	if IsDNSEnabled() {
+		err = stopAndRemoveServiceContainer(tenantID)
 		if err != nil {
 			log.Errorf("Error in stopping service container for tenant: %+v", tenantID)
 			return err
 		}
 	}
 
-	return nil
+	return err
```

Comment:
I don't think err needs to be a function scope here. It can be just block scope.